app/src/main/java/com/example/coopertest/TestActivity.kt

```diff
@@ -164,7 +172,13 @@ class TestActivity : AppCompatActivity(), OnMapReadyCallback {
     }
 
     private fun formatSpeed(speed: Float): String {
-        return "%.1f m/s".format(speed)
+        val mSharedPreference = PreferenceManager.getDefaultSharedPreferences(objectContext.getContext())
+        miles = mSharedPreference.getBoolean("miles", false)
+        if (!miles){
```

Comment:
This is nice, but I think we should have a Speed class that would do this formatting. What do you think?

Reply:
I think we use this just one time, so I don't know if it's a good idea to create a class, but create a function I a good idea I think. I will do this.